openpdf/src/main/java/com/lowagie/text/pdf/PdfCopy.java

```diff
@@ -489,17 +489,6 @@ public void close() {
         }
     }
 
-    public PdfIndirectReference add(PdfOutline outline) {
-        return null;
-    }
-
-    public void addAnnotation(PdfAnnotation annot) {
-    }
-
-    PdfIndirectReference add(PdfPage page, PdfContents contents) throws PdfException {
-        return null;
-    }
-
```

Comment:
ok, so addAnnotation(PdfAnnotation) and add(PdfPage, PdfContent) are not overwriting the methods of PdfWriter anymore, and so the methods of PdfWriter would be called.
I see that add(PdfOutline) has no parent method. And deleting if could be ok, as the method doesn't do anything.

I'm just not sure, if allowing PdfCopy to manipulate the content of the document is what the original author planed for the class.

Reply:
> I'm just not sure, if allowing PdfCopy to manipulate the content of the document is what the original author planed for the class.

You can already do the exact same thing:

```
PdfCopy copy = new PdfSmartCopy(document, outputStream)) {
document.open();
document.add(new Paragraph("Front page"));
document.newPage();
```
should do the exact same thing as the much more verbose and complicated
```
PdfCopy copy = new PdfSmartCopy(document, outputStream)) {
document.open();

ByteArrayOutputStream baos = new ByteArrayOutputStream();
Document tempDocument = new Document();
PdfWriter tempWriter = PdfWriter.getInstance(tempDocument, baos);
tempDocument.open();
tempDocument.add(new Phrase("Front page"));
tempDocument.close();
PdfReader tempReader = new PdfReader(baos.toByteArray());
PdfImportedPage page = copy.getImportedPage(tempReader, 1);
copy.addPage(page);
copy.freeReader(tempReader);
```